suite-native/discovery/src/discoveryMiddleware.ts

```diff
@@ -0,0 +1,76 @@
+import { isAnyOf } from '@reduxjs/toolkit';
+
+import {
+    authorizeDevice,
+    deviceActions,
+    selectDevice,
+    accountsActions,
+    createDiscoveryThunk,
+    startDiscoveryThunk,
+    discoveryActions,
+    selectIsDeviceReadyForDiscovery,
+    selectIsDeviceDiscoveryActive,
+} from '@suite-common/wallet-core';
+import { createMiddlewareWithExtraDeps } from '@suite-common/redux-utils';
+
+const isDiscoveryStartingAction = isAnyOf(
+    deviceActions.selectDevice,
+    deviceActions.authDevice,
+    accountsActions.changeAccountVisibility,
+);
+export const prepareDiscoveryMiddleware = createMiddlewareWithExtraDeps(
+    async (action, { dispatch, next, getState }) => {
+        if (deviceActions.forgetDevice.match(action) && action.payload.state) {
+            dispatch(discoveryActions.removeDiscovery(action.payload.state));
+        }
+
+        const previousDevice = selectDevice(getState());
+        await next(action); // pass the action
+        const nextDevice = selectDevice(getState());
+
+        const isDeviceNewlyConnected = !!(!previousDevice?.connected && nextDevice?.connected);
+        const isDeviceNewlyAcquired = !!(!previousDevice?.features && nextDevice?.features);
```

Comment:
Do you need the `!!` at the begining? I guess this could be simplified quite easily like this:
`const isDeviceNewlyConnected = !previousDevice?.connected && nextDevice?.connected;
const isDeviceNewlyAcquired = !previousDevice?.features && nextDevice?.features;`

Reply:
Yes, the conversion to the boolean is needed. Because of the way how is the `&&` operator evaluated. It results in the left-most falsy value found or if are all the values truthy, then it results in the content of the last value. 

```ts
const isDeviceNewlyConnected = !previousDevice?.connected && nextDevice?.connected;
```

would be type of `boolean | undefined`

and 
```ts
const isDeviceNewlyAcquired = !previousDevice?.features && nextDevice?.features;
```

would result in type `false | Features | undefined`

Reply:
nit: Quite a lot of `!!` for readability purposes I would suggest use `Boolean(...)` instead of that `!!` at start

Reply:
Changed to `Boolean` in 620fca7ba.